app/models/comment.rb

```diff
@@ -0,0 +1,9 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, counter_cache: true
+
+  def self.update_post_comments_counter(comment)
+    post = comment.post
+    post.update(comments_count: post.comments.count)
+  end
```

Comment:
- [ ] It's more easily doing it this way:

```
  def update_post_comments_counter(comment)
    post.increment!(:comments_counter)
  end
```

It seems you use comments_counter in the table not comments_count. Always double check columns in the db